pixeltable/globals.py

```diff
@@ -57,22 +58,56 @@ def _handle_path_collision(
     return None
 
 
+if TYPE_CHECKING:
+    import os
+    from pathlib import Path
+
+    import datasets  # type: ignore[import-untyped]
+
+    RowData = list[dict[str, Any]]
+    TableDataSourceType = Union[
+        str,
+        os.PathLike,
+        Path,  # OS paths, filenames, URLs
+        Iterator[dict[str, Any]],  # iterator producing dictionaries of values
+        RowData,  # list of dictionaries
+        DataFrame,  # Pixeltable DataFrame
+        pd.DataFrame,  # pandas DataFrame
+        'datasets.Dataset',
+        'datasets.DatasetDict',  # Huggingface datasets
+    ]
+
+
 def create_table(
     path_str: str,
-    schema_or_df: Union[dict[str, Any], DataFrame],
+    schema: Union[dict[str, Any], None] = None,
     *,
+    source: Optional[TableDataSourceType] = None,
+    source_format: Optional[Literal['csv', 'excel', 'parquet', 'json']] = None,
+    schema_overrides: Optional[dict[str, Any]] = None,
+    on_error: Literal['abort', 'ignore'] = 'abort',
     primary_key: Optional[Union[str, list[str]]] = None,
     num_retained_versions: int = 10,
     comment: str = '',
     media_validation: Literal['on_read', 'on_write'] = 'on_write',
     if_exists: Literal['error', 'ignore', 'replace', 'replace_force'] = 'error',
+    **kwargs: Any,  # Explicit arguments to source data provider
 ) -> catalog.Table:
     """Create a new base table.
```

Comment:
The docstring would benefit from more elaboration now. In particular, I'm envisioning a clear description of the semantics and allowable call patterns of schema/source/etc.